activerecord/test/cases/adapter_test.rb

```diff
@@ -717,6 +717,27 @@ def teardown
         end
       end
 
+      test "query cacheable idempotent SELECT queries allow retries" do
+        @connection.enable_query_cache!
+
+        notifications = capture_notifications("sql.active_record") do
+          assert (a = Author.first)
+          assert Post.where(id: [1, 2]).first
+          assert Post.find(1)
+          assert Post.find_by(title: "Welcome to the weblog")
```

Comment:
```suggestion
          assert_not_nil (a = Author.first)
          assert_not_nil Post.where(id: [1, 2]).first
          assert_not_nil Post.find(1)
          assert_not_nil Post.find_by(title: "Welcome to the weblog")
```

Would produce a much better error message in case this ever fails.